src/main/java/org/elasticsearch/script/CompiledScript.java

```diff
@@ -24,20 +24,42 @@
  */
 public class CompiledScript {
 
+    private final ScriptService.ScriptType type;
+    private final String name;
     private final String lang;
     private final Object compiled;
 
     /**
      * Constructor for CompiledScript.
+     * @param type The type of script to be executed.
+     * @param name The name of the script to be executed.
      * @param lang The language of the script to be executed.
      * @param compiled The compiled script Object that is executable.
      */
-    public CompiledScript(String lang, Object compiled) {
+    public CompiledScript(ScriptService.ScriptType type, String name, String lang, Object compiled) {
```

Comment:
I think it probably does.  I will opt to leave this for another PR.  We may want to look at lot of the smaller classes in ScriptService as well.  Indexed script for instance is only used in a single place as far as I can tell, so maybe we convert that to a method to get the indexed script name out of the url?